src/odemis/acq/test/fastem_test.py

```diff
@@ -435,5 +436,84 @@ def on_progress_update(self, future, start, end):
         self.updates += 1
 
 
+class TestFastEMAcquisitionTask(unittest.TestCase):
+    """Test methods of the fastem.AcquistionTask class."""
+
+    @classmethod
+    def setUpClass(cls):
+        if TEST_NOHW:
+            # If we are testing without hardware we just need a few attributes to be set correctly.
+            cls.scanner = None
+            cls.asm = None
+
+            # Use Mocks of the classes to be able to call the fake VAs as for instance mppc.dataContent.value
+            cls.mppc = Mock()
+            cls.mppc.dataContent.value = 'empty'
+
+            cls.multibeam = Mock()
+            cls.multibeam.pixelSize.value = (4.0e-9, 4.0e-9)
+            cls.multibeam.resolution.value = (800, 800)
+
+            cls.descanner = None
+            cls.stage = None
+            cls.ccd = None
+            cls.beamshift = None
+            cls.lens = None
+        else:
+            # get the hardware components from the simulators or hardware
+            cls.scanner = model.getComponent(role='e-beam')
+            cls.asm = model.getComponent(role="asm")
+            cls.mppc = model.getComponent(role="mppc")
+            cls.multibeam = model.getComponent(role="multibeam")
+            cls.descanner = model.getComponent(role="descanner")
+            cls.stage = model.getComponent(
+                role="stage")  # TODO replace with stage-scan when ROA conversion method available
+            cls.ccd = model.getComponent(role="diagnostic-ccd")
+            cls.beamshift = model.getComponent(role="ebeam-shift")
+            cls.lens = model.getComponent(role="lens")
+
+    def test_get_abs_stage_movement(self):
+        """Test the correct initial and final stage position is returned for ROAs of different field sizes."""
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+
+        # Loop over different x and y field sizes.
+        for x_fields, y_fields in zip((1, 2, 40, 34, 5), (1, 22, 43, 104, 25)):
+            # The coordinates of the ROA in meters.
+            xmin, ymin, xmax, ymax = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)
+            coordinates = (xmin, ymin, xmax, ymax)  # in m
+
+            # Create an ROC, because it is needed to initialize the ROA.
+            roc = fastem.FastEMROC("roc_name", coordinates=(0, 0, 1, 1))
+
+            # Create an ROA with the coordinates of the field.
+            roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+            roa = fastem.FastEMROA(roa_name, coordinates, roc,
+                                   self.asm, self.multibeam, self.descanner,
+                                   self.mppc
+                                   )
```

Comment:
Do you have a reason why I need to do this? As far as I know this is just an acceptable coding style. So is it because it's consistent with the rest of this file, or do you have some resource that says this is the better way?

Reply:
If I read this source correctly it is not PEP8 standard compliant to put the opening bracket in line with the arguments, but then the closing one not. I think either both brackets should be placed in their own line, or both in line with the arguments. 
https://blog.finxter.com/pep-8-hanging-indentation-and-closing-brackets-in-python/ 
https://www.python.org/dev/peps/pep-0008/#indentation